src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -62,7 +73,30 @@ class EntityClient (requestContext: RequestContext) extends Actor {
         requestContext.failWith(
           new RequestProcessingException(StatusCodes.InternalServerError, error.getMessage))
     }
+  }
 
+  def createEntities(wsNamespace: String, wsName: String, entities: Seq[Entity]) = {
+    val results = entities.map(entity => createEntityOrReportError(wsNamespace, wsName, entity))
+    // have to manually construct an HttpResponse here because we're aggregating the results of multiple calls to Rawls
+    val response = HttpResponse(OK, HttpClient.createJsonHttpEntity(results.toJson.compactPrint))
+    requestContext.complete(response)
+  }
+
+  def createEntityOrReportError(workspaceNamespace: String, workspaceName: String, entity: Entity): EntityCreateResult = {
+    val url = s"${FireCloudConfig.Workspace.entityPathFromWorkspace(workspaceNamespace, workspaceName)}"
+    val externalRequest = Post(url, HttpClient.createJsonHttpEntity(entity.toJson.compactPrint))
+    val pipeline = addHeader(Cookie(requestContext.request.cookies)) ~> sendReceive
+    // TODO add a timeout?
```

Comment:
Time-outs should be handled at the app layer - I'd be hesitant to add them here unless *really* needed.